### DISCUSSION TITLE
Always Keep Floating Windows On Top - Define workspace layouts

### DISCUSSION THREAD

**Discussion by stottm** [nikitabobko/AeroSpace#713] [questions-and-answers] [answered]
It's possible for a floating window to slide behind the tiled windows.  If there is a way to prevent it? But I would expect it to the be the default behavior because who wants a floating window to hide behind another window? This may be due to the fact that floating windows are part of the layout tree.  

I would also like to define workspaces where all windows float while other workspaces tile.

Other tiling managers have the concept of a scratchpad that can slide in or fade in on-top of your tiling windows and you can stick things like widgets and stuff on the scratchpad. Then show or hide the scratchpad as needed. 

I see some preliminary consideration for sticky windows (available on all workspaces) that would be nice to have as well.

There appears to be a bug where some windows (Arc Browser) may be moved to the furthest right on the layout tree.  This usually happens when exiting full screen, etc. The window then moves on it's own.  Keep my windows where I positioned them and don't re-sort them.

**Comment by jasoncorso** [comment]
:+1: for the floats should be always on top point

**Comment by derpwiz** [comment]
100% was looking for this functionality, I would love to see it

**Comment by tomlidobnik** [comment]
Would it be possible to hide all tiled windows? For example if I had my browser tiled and I opened a floating window, if I could hide tiled windows so that I can focus the floating window behind.

**Comment by jonathf** [comment]
I am using [choose](https://github.com/chipsenkbeil/choose) as a dmenu/rofi replacement. Works great until it sometimes hides itself underneath other windows. Forcing always-on-top for it would save me the pain of having to search out an empty workspace, cancel choose, move back to where I wanted to use it, and trigger it again.

**Reply by techeddie** [reply]
Let's hope that feature will be available soon.

**Comment by nikitabobko** [comment, marked as answer]
https://github.com/nikitabobko/AeroSpace/issues/4

**Reply by nikitabobko** [reply]
But it's probably not possible with public APIs

**Comment by porkcharsui** [comment]
I believe this may fix the issue I'm seeing when using DaVinci Resolve.  In the attached video I am play a video from the editor window and then toggle Resolve's "full view" playback. For some reason, an app window I used prior to switching to Resolve will pop on top of the "full view" video playing back in Resolve. 

Note that when I toggle Apple's "full screen" mode on the Resolve editor window and then toggle "full view" playback, this issue does not occur. 

My aerospace config is currently set to this:
```
[[on-window-detected]] 
    if.app-id = 'com.blackmagic-design.DaVinciResolve'
    check-further-callbacks = false
    run = ['layout floating'] 
```

I did a test running `aerospace list-windows --all` in a loop to see what the full screen video windows identified as while the issue occurs. `128` is the full screen video and `9209` was the editing UI:

```
9209  | DaVinci Resolve  | Untitled [Project](url)
128   | DaVinci Resolve  |
```

https://github.com/user-attachments/assets/97b3fe01-2969-4e95-b1db-3595b808a51a
